digdag-core/src/main/java/io/digdag/core/agent/RequireOperatorFactory.java

```diff
@@ -65,42 +68,157 @@ private RequireOperator(OperatorContext context, TaskCallbackApi callback)
+
+            Optional<StoredSessionAttempt> attempt = Optional.absent();
+            Optional<SessionAttemptConflictException> sessionAttemptConflictException = Optional.absent();
+            Optional<ProjectIdentifier> projectIdentifier = Optional.absent();
+            /**
+             *  First of all, try to start attempt by startSession()
+             *  If no attempt exists (no conflict), it return new StoredSessionAttempt.
+             *    - set state param "require_kicked" to true.
+             *    - task is retried to wait for done by nextPolling.
+             *  If something errors happen, it will throw following exceptions.
+             *    - ResourceNotFoundException ... workflow ,project name(or id) are wrong. -> processed as deterministic error
+             *    - ResourceLimitExceededException ... this exception should be deterministic error
+             *    - SessionAttemptConflictException ... this is not error.
+             *      - If the conflict attempt is still running, wait for until done by nextPolling.
+             *      - If done, check the state param "require_kicked" and whether the attempt is kicked by this require> or not.
+             *        - If not kicked by this require>, check result and rerun_on option and determine rerun or not.
+             *          - if need to rerun, generate unique retry_attempt_name and set to "rerun_on_retry_attempt_name"
+             *          - throw nextPolling and in next call of runTask(), "rerun_on_retry_attempt_name" is used as retry_attempt_name and must succeed to create new attempt because it is unique.
+             *        - For both kicked and not kicked, check the result and ignore_failure option
+             *          - If ignore_failure is true or attempt finished successfully, require> op finished successfully
+             *          - else finished with exception.
+             */
             try {
-                StoredSessionAttempt attempt = callback.startSession(
+                projectIdentifier = Optional.of(makeProjectIdentifier());
+
+                attempt = Optional.of(callback.startSession(
                         context,
                         request.getSiteId(),
-                        projectId,
+                        projectIdentifier.get(),
                         workflowName,
                         instant,
                         retryAttemptName,
-                        overrideParams);
+                        overrideParams));
+            }
+            catch (SessionAttemptConflictException ex) {
+                sessionAttemptConflictException = Optional.of(ex);
+            }
+            catch (ResourceNotFoundException ex) {
+                throw new TaskExecutionException(String.format(ENGLISH, "Dependent workflow does not exist. %s, workflowName:%s",
+                        projectIdentifier.transform((p)->p.toString()).or(""), workflowName ));
```

Comment:
There is an unexpected space between `workflowName` and `)`.

Reply:
[minor] `(p)->p.toString()` can be replaced with `ProjectIdentifier::toString`

Reply:
Thanks! I will fix them.

Reply:
Fixed it. Thanks!